docs/intro/intro_symbolic_math.rst

```diff
@@ -4,96 +4,187 @@
-  by adding the two models, :code:`x + bqm`. This adds the variable labeled ``'x'``
-  in the single-variable BQM, ``x`` to model ``bqm``. You cannot add ``x`` to a
-  model---as though it were variable ``'x'``---by doing :code:`bqm.add_variable(x)`.
-
-Example: BQM
-============
-
-This example creates the BQM :math:`x + 2y -xy`:
-
->>> from dimod import Binary
->>> x = Binary('x')
->>> y = Binary('y')
->>> bqm = x + 2*y - x*y
->>> print(bqm.to_polystring())
-x + 2*y - x*y
-
-Example: CQM
-============
-
-This example uses symbolic math to set an objective (:math:`2i - 0.5ij + 10`)
-and constraints (:math:`xj <= 3` and :math:`i + j >= 1`) in a simple CQM.
-
->>> from dimod import Binary, Integer, ConstrainedQuadraticModel
->>> x = Binary('x')
->>> i = Integer('i')
->>> j = Integer('j')
->>> cqm = ConstrainedQuadraticModel()
->>> cqm.set_objective(2*i - 0.5*i*j + 10)
->>> cqm.add_constraint(x*j <= 3)                   # doctest: +IGNORE_RESULT
->>> cqm.add_constraint(i + j >= 1)                 # doctest: +IGNORE_RESULT
-
-Example: Adding Models
-======================
-
-This example uses the performant :func:`~dimod.binary.quicksum` on
-:func:`~dimod.binary.BinaryArray` to add multiple models.
-
->>> import numpy as np
->>> from dimod import BinaryArray, quicksum
-...
->>> num_vars = 10; max_bias = 5
->>> var_labels = range(num_vars)
-...
->>> models = BinaryArray(var_labels)*np.random.randint(0, max_bias, size=num_vars)
->>> x = quicksum(models)
+
+  An important distinction is that :code:`x = dimod.Binary('x')`, for example,
+  instantiates a model with a variable label ``'x'`` and not a free-floating variable
+  labeled ``x``. Consequently, you can add ``x`` to another model by adding the two
+  models,
+
+  >>> bqm = dimod.BinaryQuadraticModel('BINARY')
+  >>> bqm += x
+
+  which adds the variable labeled ``'x'`` in the single-variable BQM, ``x``, to
+  model ``bqm``. You cannot add ``x`` to a model---as though it were variable
+  ``'x'``---by doing :code:`bqm.add_variable(x)`.
+
+Representing Constraints
+========================
+
+Many real-world problems include constraints. Typically constraints are either
+equality or inequality, in the form of a left-hand side(``lhs``), right-hand-side
+(``rhs``), and the :class:`dimod.sym.Sense` (:math:`\le`, :math:`\ge`, or
+:math:`==`). For example, the constraint of the rectangle problem above,
+
+.. math::
+
+  \textrm{s.t.} \quad 2i+2j \le P
+
+has a ``lhs`` of :math:`2i+2j` less or equal to a ``rhs`` of a some real number
+(:math:`8`):
+
+>>> print(constraint.lhs.to_polystring(), constraint.sense.value, constraint.rhs)  # doctest:+SKIP
+2*i + 2*j <= 8
+
+You can create such an equality or inequality symbolically, and it is shown
+with the model:
+
+>>> print(type(3.75 * i <= 4))
+<class 'dimod.sym.Le'>
+>>> 3.75 * i <= 4
+QuadraticModel({'i': 3.75}, {}, 0.0, {'i': 'INTEGER'}, dtype='float64') <= 4
+
+Performance
```

Comment:
There is obviously a lot more to be said here, but I think it makes sense to move that into a future "How to scale your problem" page.